lib/internal/process.js

```diff
@@ -22,6 +23,65 @@ const assert = process.assert = function(x, msg) {
 };
 
 
+function setup_cpuUsage() {
+  const _cpuUsage = process.cpuUsage;
+  const cpuValues = new Uint32Array(8);
+
+  process.cpuUsage = function cpuUsage(prevValue) {
+    const err = _cpuUsage(cpuValues);
+    if (err !== 0) throw new Error('unable to obtain cpu usage time');
+
+    // the second element of user/system comes back from libuv as
+    // microseconds, but we convert to nanoseconds here (* 1000) to
+    // be consistent with hrtime()
+    var currValue = {
+      user: [
+        cpuValues[0] * 0x100000000 + cpuValues[1],
+        (cpuValues[2] * 0x100000000 + cpuValues[3]) * 1000
+      ],
+      system: [
+        cpuValues[4] * 0x100000000 + cpuValues[5],
+        (cpuValues[6] * 0x100000000 + cpuValues[7]) * 1000
```

Comment:
Personally I'd go with `seconds.microseconds`. The double value has high enough resolution, and takes less time to operate on. So I'd switch the array to a `Float64Array` and assign `rusage.ru_utime.tv_sec + rusage.ru_utime.tv_usec / 1e6`. This type of operation doesn't require nanosecond precision.